Build error when compiling for Windows from Linux

**Godot version:** Git https://github.com/godotengine/godot/commit/bde3e884c7b1506dd74cb39ea479ea98dd04e7a1

**OS/device including version:** Fedora 28, GCC 8.1.1

**Issue description:**

Compiling for Windows from Linux using MinGW fails with the following error:

```
Compiling ==> modules/websocket/lws_client.cpp
modules/websocket/lws_client.cpp: In member function 'virtual Error LWSClient::connect_to_host(String, String, uint16_t, bool, PoolVector<String>)':
modules/websocket/lws_client.cpp:87:41: error: invalid conversion from 'const char*' to 'size_t {aka long long unsigned int}' [-fpermissive]
  strncpy(abuf, addr_str.ascii().get_data(), 1023);
                ~~~~~~~~~~~~~~~~~~~~~~~~~^~
modules/websocket/lws_client.cpp:87:49: error: invalid conversion from 'int' to 'const char*' [-fpermissive]
  strncpy(abuf, addr_str.ascii().get_data(), 1023);
                                                 ^
modules/websocket/lws_client.cpp:87:49: error: too few arguments to function 'errno_t strncpy_s(char*, size_t, const char*, size_t)'
In file included from /usr/x86_64-w64-mingw32/sys-root/mingw/include/string.h:190:0,
                 from ./core/os/copymem.h:42,
                 from ./core/dvector.h:34,
                 from ./core/variant.h:41,
                 from ./core/method_ptrcall.h:36,
                 from ./core/method_bind.h:35,
                 from ./core/class_db.h:34,
                 from ./core/reference.h:34,
                 from ./core/io/stream_peer.h:34,
                 from modules/websocket/lws_helper.h:36,
                 from modules/websocket/lws_client.h:36,
                 from modules/websocket/lws_client.cpp:32:
/usr/x86_64-w64-mingw32/sys-root/mingw/include/sec_api/string_s.h:39:27: note: declared here
   _CRTIMP errno_t __cdecl strncpy_s(char *_Dst,size_t _DstSizeInChars,const char *_Src,size_t _MaxCount);
                           ^~~~~~~~~
modules/websocket/lws_client.cpp:89:38: error: invalid conversion from 'const char*' to 'size_t {aka long long unsigned int}' [-fpermissive]
  strncpy(hbuf, p_host.utf8().get_data(), 1023);
                ~~~~~~~~~~~~~~~~~~~~~~^~
modules/websocket/lws_client.cpp:89:46: error: invalid conversion from 'int' to 'const char*' [-fpermissive]
  strncpy(hbuf, p_host.utf8().get_data(), 1023);
                                              ^
modules/websocket/lws_client.cpp:89:46: error: too few arguments to function 'errno_t strncpy_s(char*, size_t, const char*, size_t)'
In file included from /usr/x86_64-w64-mingw32/sys-root/mingw/include/string.h:190:0,
                 from ./core/os/copymem.h:42,
                 from ./core/dvector.h:34,
                 from ./core/variant.h:41,
                 from ./core/method_ptrcall.h:36,
                 from ./core/method_bind.h:35,
                 from ./core/class_db.h:34,
                 from ./core/reference.h:34,
                 from ./core/io/stream_peer.h:34,
                 from modules/websocket/lws_helper.h:36,
                 from modules/websocket/lws_client.h:36,
                 from modules/websocket/lws_client.cpp:32:
/usr/x86_64-w64-mingw32/sys-root/mingw/include/sec_api/string_s.h:39:27: note: declared here
   _CRTIMP errno_t __cdecl strncpy_s(char *_Dst,size_t _DstSizeInChars,const char *_Src,size_t _MaxCount);
                           ^~~~~~~~~
modules/websocket/lws_client.cpp:91:38: error: invalid conversion from 'const char*' to 'size_t {aka long long unsigned int}' [-fpermissive]
  strncpy(pbuf, p_path.utf8().get_data(), 2047);
                ~~~~~~~~~~~~~~~~~~~~~~^~
modules/websocket/lws_client.cpp:91:46: error: invalid conversion from 'int' to 'const char*' [-fpermissive]
  strncpy(pbuf, p_path.utf8().get_data(), 2047);
                                              ^
modules/websocket/lws_client.cpp:91:46: error: too few arguments to function 'errno_t strncpy_s(char*, size_t, const char*, size_t)'
In file included from /usr/x86_64-w64-mingw32/sys-root/mingw/include/string.h:190:0,
                 from ./core/os/copymem.h:42,
                 from ./core/dvector.h:34,
                 from ./core/variant.h:41,
                 from ./core/method_ptrcall.h:36,
                 from ./core/method_bind.h:35,
                 from ./core/class_db.h:34,
                 from ./core/reference.h:34,
                 from ./core/io/stream_peer.h:34,
                 from modules/websocket/lws_helper.h:36,
                 from modules/websocket/lws_client.h:36,
                 from modules/websocket/lws_client.cpp:32:
/usr/x86_64-w64-mingw32/sys-root/mingw/include/sec_api/string_s.h:39:27: note: declared here
   _CRTIMP errno_t __cdecl strncpy_s(char *_Dst,size_t _DstSizeInChars,const char *_Src,size_t _MaxCount);
                           ^~~~~~~~~
scons: *** [modules/websocket/lws_client.windows.opt.tools.64.o] Error 1
scons: building terminated because of errors.
```

[Full build log on GitLab CI](https://gitlab.com/Calinou/godot-builds-ci/-/jobs/110198201)

This is likely a regression from https://github.com/godotengine/godot/commit/b902a2f2a7438810cdcb053568ed5c27089b1e8a#diff-ae2fb9a595568921446b5f494f3323b2.

--- a/modules/websocket/emws_server.cpp
+++ b/modules/websocket/emws_server.cpp
@@ -71,6 +71,9 @@ int EMWSServer::get_peer_port(int p_peer_id) const {
 void EMWSServer::disconnect_peer(int p_peer_id, int p_code, String p_reason) {
 }
 
+void EMWSServer::poll() {
+}
+
 EMWSServer::EMWSServer() {
 }
 

--- a/modules/websocket/lws_client.cpp
+++ b/modules/websocket/lws_client.cpp
@@ -34,10 +34,6 @@
 #include "core/io/stream_peer_ssl.h"
 #include "tls/mbedtls/wrapper/include/openssl/ssl.h"
 
-#if defined(MINGW_ENABLED) || defined(_MSC_VER)
-#define strncpy strncpy_s
-#endif
-
 Error LWSClient::connect_to_host(String p_host, String p_path, uint16_t p_port, bool p_ssl, PoolVector<String> p_protocols) {
 
 	ERR_FAIL_COND_V(context != NULL, FAILED);
@@ -80,26 +76,11 @@ Error LWSClient::connect_to_host(String p_host, String p_path, uint16_t p_port, 
 		ERR_FAIL_V(FAILED);
 	}
 
-	char abuf[1024];
-	char hbuf[1024];
-	char pbuf[2048];
-	String addr_str = (String)addr;
-	strncpy(abuf, addr_str.ascii().get_data(), 1023);
-	abuf[1023] = '\0';
-	strncpy(hbuf, p_host.utf8().get_data(), 1023);
-	hbuf[1023] = '\0';
-	strncpy(pbuf, p_path.utf8().get_data(), 2047);
-	pbuf[2047] = '\0';
-
 	i.context = context;
 	if (p_protocols.size() > 0)
 		i.protocol = _lws_ref->lws_names;
 	else
 		i.protocol = NULL;
-	i.address = abuf;
-	i.host = hbuf;
-	i.path = pbuf;
-	i.port = p_port;
 
 	if (p_ssl) {
 		i.ssl_connection = LCCSCF_USE_SSL;
@@ -109,7 +90,16 @@ Error LWSClient::connect_to_host(String p_host, String p_path, uint16_t p_port, 
 		i.ssl_connection = 0;
 	}
 
+	// This String needs to survive till we call lws_client_connect_via_info
+	String addr_str = (String)addr;
+
+	i.address = addr_str.ascii().get_data();
+	i.host = p_host.utf8().get_data();
+	i.path = p_path.utf8().get_data();
+	i.port = p_port;
+
 	lws_client_connect_via_info(&i);
+
 	return OK;
 };
 

